distributed/client.py

```diff
@@ -3199,7 +3309,9 @@ def _graph_to_futures(
         with self._refcount_lock:
             if actors is not None and actors is not True and actors is not False:
                 actors = list(self._expand_key(actors))
-
+            # if this comes from MapLayer, check if its materialized
+            if hasattr(dsk, "is_materialized"):
+                assert not dsk.is_materialized(), "Graph must be non-materialized"
```

Comment:
This might be a problem. There are unfortunatley HLGs that are materialized. We can't raise this whenever we encounter one of those. The dask/distributed test suite might not have sufficient coverage for this